3_plot/plot_doy_mean.py

```diff
@@ -0,0 +1,29 @@
+import pandas as pd
+import os
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+def plot_doy_means(out_file, df_sel_depths):
+    df_long = df_sel_depths.reset_index().melt(var_name="depth",
+                                               value_name = "temperature (C)",
+                                               id_vars=["doy", "site_id"])
+    sns.relplot(data=df_long, x="doy", y="temperature (C)", col="site_id",
+                hue="depth", col_wrap=2, kind='line')
+    plt.tight_layout()
+    plt.savefig(out_file)
+    return out_file
+
+def main(out_dir, combined_doy_means, out_file, depths):
```

Comment:
Here, `out_dir` can be inferred from `out_file` instead of passed as a separate variable.
```suggestion
def main(combined_doy_means, out_file, depths):
    out_dir = os.path.dirname(out_file)
```